src/DotNetEnv/Configuration/EnvConfigurationProvider.cs

```diff
@@ -37,22 +38,11 @@ public override void Load()
             }
 
             // Since the Load method does not take care of cloberring, We have to check it here!
-            foreach (var value in values)
-            {
-                var key = NormalizeKey(value.Key);
-                if (this.options.ClobberExistingVars)
-                {
-                    this.Data[key] = value.Value;
-                }
-                else
-                {
-                    if (!this.Data.ContainsKey(key))
-                    {
-                        this.Data.Add(key, value.Value);
-                    }
-                }
-            }
+            foreach (var value in values.ToDotEnvDictionary(options.ClobberExistingVars ? CreateDictionaryOption.TakeLast : CreateDictionaryOption.TakeFirst))
+                Data[NormalizeKey(value.Key)] = value.Value;
```

Comment:
Used ToDotEnvDictionary in here.
Tests still succeed, it should behave exactly as it was before. But investigating this I think the Clobber-behaviour at all is kinda "inconsistent".

That's not something to be solved in this PR, but there is some confusion around `SetEnvVars`, `NoClobber` and `Interpolated` for me.
Examples here, but I guess I will raise separate issues (and creating some tests to show this inconsistent behavior to proove it ;) )
- `Interpolated` will not work as expected with `SetNoEnvVars` (Interploated just checks EnvironmentVars on `GetValue`)
- `NoClobber` together with `EnvConfigurationProvider` does not check EnvironmentVars ==> it "overrides" EnvironmentVariables (in Configuration, not the EnvironmentVars itself of course), but `Interpolated` still uses the existing EnvironmentVariables

Reply:
Hmm, this is potentially a significant behavior change. Before it would calculate the key from all values and then check clobbering on the *calculated key*. Now you have it doing clobbering on the values and then calculating the key -- which could potentially lead to different values mapping to the same keys and thus extra, unexpected clobbering.

Reply:
But *if* this is not going to lead to unexpected clobbering, it is a nicely cleaned up approach. I would pull out the dict opt into its own var on the line before, for readability, tho.

Reply:
Refactored it (pushing it to an own var).

I saw this difference too, but it should be irrelevant here.
In general I would say it is better to check before normalizing, because clobbering while Parsing is not Normalizing any key too... And it should not of course, because we check the clobbering against the Keys within the DotEnvFile and the EnvironmentVariables, and they will never have DotNetConfiguration-Normalized-Keys...

Reply:
About `potentially mapping to the same keys`:
The normalizedKeys contain `:` instead of `__`. That's done because EnvVars cannot contain `:`, but that's the default separator in C#-Configs for `sections`.

I just tested it and we also do not allow `:` in env-Files (which is expected behavior), so we cannot introduce "duplicated Keys" which yould be normalized to a single one via Normalization here.

But of course this is an important discussion, so maybe we should move this whole point to a different PR, not "polluting" this one here with that change. Then we can also discuss the other points I mentioned above there.

Reply:
It's fine to make this change now, given the logic. Feel free to also add the test you used for the `:` as confirmation.

Reply:
Added a test for the identifier, but did not add any reference to that.
Guess that's not needed, but the test is a good confirmation for us to proove that the change will not break anything.

Since the actual code does not care for already existing EnvironmentVariables (just DotEnv-File Variables) it prooves that there will be no new behavior due to the change.